File: iosApp/iosAppTests/UIComponent/Order/OrderStateRobot.swift
import ComposeApp

final class OrderStateRobot: BaseStateRobot<OrderUiState, OrderEvent> {

    override func defaultState() -> OrderUiState {
        OrderUiState(
            categoryPreviews: [
                CategoryPreview(id: "burgers", name: "Burgers", firstItemImageUrl: nil, itemCount: 2),
                CategoryPreview(id: "drinks", name: "Drinks", firstItemImageUrl: nil, itemCount: 0),
            ],
            cartSummary: CartSummary(itemCount: 2, total: "$11.98"),
            eventSink: createEventSink()
        )
    }

    func stateWithNoCart() -> OrderUiState {
        OrderUiState(
            categoryPreviews: [
                CategoryPreview(id: "burgers", name: "Burgers", firstItemImageUrl: nil, itemCount: 2),
            ],
            cartSummary: nil,
            eventSink: createEventSink()
        )
    }
}

File: features/order/impl/presentation/src/androidDeviceTest/kotlin/com/jkjamies/sampleplatter/features/order/presentation/OrderStateRobot.kt
package com.jkjamies.sampleplatter.features.order.presentation

import com.jkjamies.sampleplatter.core.test.StateRobot
import com.jkjamies.sampleplatter.features.order.api.domain.CartSummary
import com.jkjamies.sampleplatter.features.order.api.domain.CategoryPreview

class OrderStateRobot : StateRobot<OrderUiState, OrderEvent>() {

    override fun defaultState() = OrderUiState(
        categoryPreviews = listOf(
            CategoryPreview(id = "burgers", name = "Burgers", firstItemImageUrl = null, itemCount = 2),
            CategoryPreview(id = "drinks", name = "Drinks", firstItemImageUrl = null, itemCount = 0),
        ),
        cartSummary = CartSummary(itemCount = 2, total = "\$12.99"),
        eventSink = createEventSink(),
    )

    fun stateWithNoCart() = defaultState().copy(
        cartSummary = null,
        eventSink = createEventSink(),
    )

    fun stateWithEmptyPreviews() = defaultState().copy(
        categoryPreviews = emptyList(),
        eventSink = createEventSink(),
    )
}
